aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1877,14 +1835,14 @@ impl AptosVM {
         func_name: Identifier,
         type_args: Vec<TypeTag>,
         arguments: Vec<Vec<u8>>,
-        gas_budget: u64,
+        max_gas_amount: u64,
     ) -> ViewFunctionOutput {
         let resolver = state_view.as_move_resolver();
         let vm = AptosVM::new(&resolver);
         let log_context = AdapterLogSchema::new(state_view.id(), 0);
-        let mut gas_meter = match Self::memory_tracked_gas_meter(&vm, &log_context, gas_budget) {
+        let mut gas_meter = match vm.make_standard_gas_meter(max_gas_amount.into(), &log_context) {
```

Comment:
unwrap_or_else and ?

Reply:
I guess you mean map_err and ?

Reply:
right)

Reply:
actually it wouldn't work, because we return a non-result type 😁 do you think it is worth changing the signature to result?